upgrade/upgrade-provider.go

```diff
@@ -331,6 +331,13 @@ func UpgradeProvider(ctx Context, repoOrg, repoName string) error {
 			"go", "get", "github.com/pulumi/pulumi-terraform-bridge/v3@"+targetBridgeVersion)).
 			In(repo.providerDir()))
 
+		steps = append(steps, step.Cmd(exec.CommandContext(ctx,
+			"go", "get", "github.com/hashicorp/terraform-plugin-framework")).
+			In(repo.providerDir()))
+		steps = append(steps, step.Cmd(exec.CommandContext(ctx,
+			"go", "get", "github.com/hashicorp/terraform-plugin-mux")).
+			In(repo.providerDir()))
```

Comment:
Experimenting shows that running `go get ...` for a dependency that is not used adds the dependency to `go.mod` and `go.sum`, upgrading shared transitive dependencies. Running `go mod tidy` after that removes the unused dependency, but doesn't restore shared transitive dependencies to their prior state. I think this is ok.

Can we add a `go mod tidy` after these steps to remove the dependency if it's unnecessary.